src/components/common/multi-image-resizer.svelte

```diff
@@ -0,0 +1,168 @@
+<script>
+  import { getContext } from 'svelte';
+  import { stores } from '@sapper/app';
+  import { Button, Dialog, Modal, Divider, Loading } from 'attractions';
+  import { snackbarContextKey } from 'attractions/snackbar';
+  import Cropper from 'svelte-easy-crop';
+  import * as api from 'src/utils/api.js';
+
+  const { session } = stores();
+
+  export let aspectRatio = 1;
+  export let open = false;
+
+  let currentImage = 0;
+  let images = [];
+
+  const crop = { x: 0, y: 0 };
+
+  function saveBounds({ detail }) {
+    images[currentImage].pixels = detail.pixels;
+  }
+
+  function resetCropArea() {
+    crop.x = 0;
+    crop.y = 0;
+  }
+
+  async function repositionCursor() {
+    for (let idx = 0; idx < images.length; ++idx) {
+      if (images[idx].uploadPromise == null) {
+        currentImage = idx;
+        resetCropArea();
+        return;
+      }
+    }
+
+    try {
+      currentImage = null;
+      await Promise.all(images.map(image => image.uploadPromise));
+    } finally {
+      open = false;
+      images = [];
+      currentImage = 0;
+    }
+  }
+
+  function dismissImage(image = null) {
+    image.reject();
+    images = images.filter(thatImage => thatImage !== image);
+    repositionCursor();
+  }
+
+  async function applyCrop() {
+    const image = images[currentImage];
+    const formData = new FormData();
+    formData.append('file', image.file);
+    formData.append('x', image.pixels.x);
+    formData.append('y', image.pixels.y);
+    formData.append('width', image.pixels.width);
+    formData.append('height', image.pixels.height);
+
+    try {
+      image.uploadPromise = api.json(api.post('/file', {
+        data: formData,
+        csrfToken: $session.account.csrf_token,
+      }));
+      repositionCursor();
+      image.file._id = (await image.uploadPromise).id;
+      image.resolve();
+    } catch (e) {
+      showSnackbar({ props: { text: 'Failed to upload image, try reloading the page' } });
+      console.error(e);
+      dismissImage(image);
+    }
+  }
+
+  export function add(image) {
+    const imageObject = {
+      file: image,
+      url: URL.createObjectURL(image, { type: image.type }),
+      uploadPromise: null,
+      resolve: null,
+      reject: null,
+    };
+
+    images.push(imageObject);
+    images = images;
+
+    return new Promise((resolve, reject) => {
+      imageObject.resolve = resolve;
+      imageObject.reject = reject;
+    });
+  }
+
+  function alertOnClose({ detail }) {
+    if (!detail.value) {
```

Comment:
```suggestion
    if (detail.value) return;
```

Reply:
I'm always hesitant to accept suggestions like this, but this time especially. `detail.value` is the `open` value of the modal, and this way of writing it makes it clear that we're only acting upon the _close_ event

Reply:
`function alertOnClose({ detail: { value: open } })`, followed by `if(!open) return;` in the body should make it clear enough

Reply:
Yeah, but is all of this mess worth that tiny indent reduction? I don't think so
I know about the cognitive complexity, but I don't take it seriously in code like this (as opposed to complex view logic on the backend where it all does add up)

Reply:
And even if not for the guard clause, the rename makes it clearer